src/message_bus.zig

```diff
@@ -1020,20 +1017,18 @@ pub fn MessageBusType(comptime IO: type, comptime process_type: vsr.ProcessType)
 
             switch (connection.peer) {
                 .unknown => {},
-                .client, .client_likely => |client_id| switch (process_type) {
-                    .replica => {
-                        // A newer client connection may have replaced this one:
-                        if (bus.process.clients.get(client_id)) |existing_connection| {
-                            if (existing_connection == connection) {
-                                assert(bus.process.clients.remove(client_id));
-                            }
-                        } else {
-                            // A newer client connection may even leapfrog this connection
-                            // and then be terminated and set to null before we can get
-                            // here.
+                .client, .client_likely => |client_id| {
+                    // A newer client connection may have replaced this one:
+                    if (bus.clients.get(client_id)) |existing_connection| {
+                        assert(bus.process == .replica);
```

Comment:
Can this assert move up out of the branch, since clients don't talk to other clients?